syntax = "proto3";

service CustomerService {
    rpc CreateCustomer (CustomerRequest) returns (CustomerResponse) {}
    rpc GetCustomer (CustomerRequest) returns (CustomerResponse) {}
    rpc UpdateCustomer (CustomerRequest) returns (CustomerResponse) {}
    rpc DeleteCustomer (CustomerRequest) returns (CustomerResponse) {}
}

message CustomerRequest {
    string id = 1;
    string name = 2;
    string email = 3;
}

message CustomerResponse {
    string id = 1;
    string name = 2;
    string email = 3;
}